### ISSUE TITLE
A11Y - Incorrect voice-over focus for used saved card button for credit card autofill

### ISSUE THREAD

**Issue by nbhasin2** [mozilla-mobile/firefox-ios#15325]
## Steps to reproduce
- Foxfooding Bug: 6/30/2023 16:09:56
- No particular steps
- Foxfooding Notes
- Try to perform a remember your card flow for credit card
- (a11y S2 build 9000) When there is at least one saved card, the "Use Saved Card" control which is added to the numeric keyboard when an autocomplete=card field has focused which is not accessible with Switch Control (it does not receive focus in Item mode and the user would require tediously use gliding cursor mode to try to activate it) and it is missing a role when announced with VoiceOver. 
-- 1. It's likely that adding a trait of button to the control would allow a screen reader user to be aware that this is, in fact, an actionable item; a Switch Control to provide focus to the control; and Voice Control software to be able to call it by either name (on-screen text) or role to activate it
-- 2. Ensure the control is included in the focus and reading order in the same order as visual reading order goes, for LTR like English it's right after the "^/v" controls and before the "Done" button. At the moment VoiceOver focuses it after the "Done" which is confusing for sighted screen reader users (those who are with low vision, users with dyslexia, etc)

### Expected behavior
- See points 1 and 2 above for how the message should be announced for voice over correct focus

### Actual behavior
-  When there is at least one saved card, the "Use Saved Card" control which is added to the numeric keyboard when an autocomplete=card field has focused which is not accessible with Switch Control

### Device & build information
* Device: iPhone
* Build version: 32291
* First seen version: ? 
### Notes
Attachments: 
| Img1      | img2 |
| ----------- | ----------- |
| <img width="397" alt="image" src="https://github.com/mozilla-mobile/firefox-ios/assets/8919439/ed759c56-4971-44ab-a418-c6dc75dfbf65">      | <img width="402" alt="image" src="https://github.com/mozilla-mobile/firefox-ios/assets/8919439/3a4d7c0f-f965-4f35-a80f-e2aee0c1c3be">       |

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-6884)

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Verified as fixed on v117 (33597) with iPhone 13 Pro (15.7.1).

I was able to access used saved card and done buttons without problem and the voice-over was correctly spoken.

Regarding switch-control should use saved card and done button be accessible via switch-control? Right now this is not the case as  they cannot be accessed with the switch-control arrows. Stefan Vladutcan you confirm that this behavior is the correct one?
